### DISCUSSION TITLE
The dynamic name of the target.

### DISCUSSION THREAD

**Discussion by Art-Che** [AutomaApp/automa#1007] [Q&A]
Hello! Automa is a brilliant thing!
I can't resolve the issue:
In my CRM, when creating a contact, the id of the field that I need to change changes every time.

`#iframe_pfsowl5ko1 |> #UF_CRM_1652369033_control_VU5Sxk1670674193 > .main-ui-control`

**1670674193** this part of the id changes every time

Automa does not "see" this field when creating a new contact. What can be done?

**Comment by Kholid060** [comment]
You can exclude the element id from the selector by opening the selector settings.

![image](https://user-images.githubusercontent.com/22908993/206856100-b0417363-c580-49f4-ad58-7affbc696549.png)

**Reply by Art-Che** [reply]
Excellent! Works! Thank you very much, you are amazing!

**Comment by Art-Che** [comment]
Hi!
Again a dynamic field name.
Again a dynamic field name. I have tried all the "Selector settings"

div:nth-child(263) iframe |> div:nth-child**(347)** span:nth-child(1) > span:nth-child(2)

div:nth-child(263) iframe |> div:nth-child**(342)** span:nth-child(1) > span:nth-child(2)

Please tell me what can be done?

**Reply by Kholid060** [reply]
Can you right at the element, select the "inspect" menu, and screenshot the chrome dev tools

**Comment by Art-Che** [comment]
So?
![1](https://user-images.githubusercontent.com/120256667/206948543-1a207197-7b47-4ffe-93af-e472f86935f0.png)

**Reply by Kholid060** [reply]
Try this `#iframe_pfsowl5ko1 |> [id^="UF_CRM_"] > .main-ui-control`

**Comment by Art-Che** [comment]
![image](https://user-images.githubusercontent.com/120256667/206963304-49ee5bca-514e-4956-9255-697b94491760.png)
Error.
As far as I understand: CRM in each new company / contact / deal card generates a new id for each field.
Only the name of the field remains unchanged.
What can be done to select the field name and menu item?

I tried like this
![image](https://user-images.githubusercontent.com/120256667/206963851-91deb463-604d-4111-9e88-8d03d9a06132.png)

![image](https://user-images.githubusercontent.com/120256667/206963962-33b7141a-deb8-4333-8a7f-d7827b34cf25.png)

Got an error
![image](https://user-images.githubusercontent.com/120256667/206964048-9670608f-5dbb-463c-ae6e-a61152185a04.png)

**Reply by Kholid060** [reply]
It can't find the element because it's inside an iframe. You need to follow this format if you want to select an element inside an iframe `.iframe-selector |> .element-inside-iframe`

**Reply by Art-Che** [reply]
Thank you very much for spending time with me.
There was an error again. I don't know JS.
![image](https://user-images.githubusercontent.com/120256667/206969525-ea9d68b5-9dad-497d-bf86-b56d6e6b8661.png)
